[Python] NameError in imported module

This might be related to #2915.

Given the following 3 files:
`ex.py`:
```python3
import ll
from ll import Foo

foo_instance: Foo = ll.create("Foo")
```
`ll.py`:
```python3
from md import Foo, Bar

def create(s: str) -> Foo | Bar:
    if s == "Foo":
        return Foo(s)
    elif s == "Bar":
        return Bar(s)
    else:
        raise ValueError("Invalid class name")
```
`md.py`:
```python3
class Foo:
    def __init__(self, s: str) -> None:
        self.s = s

class Bar:
    def __init__(self, s: str) -> None:
        self.s = s
```
ESBMC produces a NameError:
```bash
$ esbmc ex.py 
ESBMC version 7.11.0 64-bit x86_64 linux
Target: 64-bit little-endian x86_64-unknown-linux with esbmclibc
Parsing ex.py

Type checking warning:
ex.py:2: error: Module "ll" does not explicitly export attribute "Foo"  [attr-defined]
ex.py:4: error: Incompatible types in assignment (expression has type "Foo | Bar", variable has type "Foo")  [assignment]
Found 2 errors in 1 file (checked 1 source file)

Converting
WARNING: Union with no recognized types, defaulting to pointer
WARNING: Undefined function: Foo
WARNING: Undefined function: Bar
ERROR: NameError: name 'Foo' is not defined
```
ESBMC version: 97828b786

--- a/regression/python/github_2960/ll.py
+++ b/regression/python/github_2960/ll.py
@@ -1,0 +1,9 @@
+from md import Foo, Bar
+
+def create(s: str) -> Foo | Bar:
+    if s == "Foo":
+        return Foo(s)
+    elif s == "Bar":
+        return Bar(s)
+    else:
+        raise ValueError("Invalid class name")

--- a/regression/python/github_2960/main.py
+++ b/regression/python/github_2960/main.py
@@ -1,0 +1,4 @@
+import ll
+from ll import Foo
+
+foo_instance: Foo = ll.create("Foo")

--- a/regression/python/github_2960/md.py
+++ b/regression/python/github_2960/md.py
@@ -1,0 +1,7 @@
+class Foo:
+    def __init__(self, s: str) -> None:
+        self.s = s
+
+class Bar:
+    def __init__(self, s: str) -> None:
+        self.s = s

--- a/regression/python/github_2960/test.desc
+++ b/regression/python/github_2960/test.desc
@@ -1,0 +1,4 @@
+CORE
+main.py
+
+^VERIFICATION SUCCESSFUL$

--- a/regression/python/version/test.desc
+++ b/regression/python/version/test.desc
@@ -1,4 +1,4 @@
-THOROUGH
+KNOWNBUG
 main.py
 
-^ERROR: Object \"importlib\.metadata\" not found.$
+^VERIFICATION SUCCESSFUL$

--- a/src/python-frontend/parser.py
+++ b/src/python-frontend/parser.py
@@ -242,28 +242,41 @@ def emit_module_json(
 
 
 def process_collected_imports(output_dir):
-    for module_name, import_info in list(module_imports.items()):
-        imported_elements = None if import_info['import_all'] \
-            else [ast.alias(name, None) for name in import_info['specific_names']]
+    # Keep processing until no new modules are discovered
+    processed_modules = set()
 
-        # Attempt to resolve and emit JSON for imported submodules (e.g., "pkg.sub")
-        if import_info['specific_names']:
-            for name in list(import_info['specific_names']):
-                emit_module_json(f"{module_name}.{name}", output_dir)
+    while True:
+        # Get modules that haven't been processed yet
+        modules_to_process = set(module_imports.keys()) - processed_modules
 
-        # Emit the module itself
-        filename = resolve_module_file(module_name, output_dir)
-        if not filename:
-            continue
+        if not modules_to_process:
+            break
 
-        # Parse once, fix relative imports, collect nested imports
-        tree = parse_file(filename)
-        for subnode in ast.walk(tree):
-            if isinstance(subnode, (ast.Import, ast.ImportFrom)):
-                rewrite_relative_import(subnode, module_name)
-                process_imports(subnode, output_dir)
+        for module_name in modules_to_process:
+            import_info = module_imports[module_name]
+            processed_modules.add(module_name)
 
-        generate_ast_json(tree, filename, imported_elements, output_dir, module_qualname=module_name)
+            imported_elements = None if import_info['import_all'] \
+                else [ast.alias(name, None) for name in import_info['specific_names']]
+
+            # Attempt to resolve and emit JSON for imported submodules (e.g., "pkg.sub")
+            if import_info['specific_names']:
+                for name in list(import_info['specific_names']):
+                    emit_module_json(f"{module_name}.{name}", output_dir)
+
+            # Emit the module itself
+            filename = resolve_module_file(module_name, output_dir)
+            if not filename:
+                continue
+
+            # Parse once, fix relative imports, collect nested imports
+            tree = parse_file(filename)
+            for subnode in ast.walk(tree):
+                if isinstance(subnode, (ast.Import, ast.ImportFrom)):
+                    rewrite_relative_import(subnode, module_name)
+                    process_imports(subnode, output_dir)
+
+            generate_ast_json(tree, filename, imported_elements, output_dir, module_qualname=module_name)
 
 
 def rewrite_relative_import(node, parent_module: str | None):

